stdlib/task/src/main/java/org/ballerinalang/stdlib/task/utils/Utils.java

```diff
@@ -0,0 +1,135 @@
+                throw new SchedulingException("AppointmentData \"" + record.stringValue() + "\" is invalid.");
+            }
+        } else {
+            cronExpression = record.stringValue();
+            if (!isValidExpression(cronExpression)) {
+                throw new SchedulingException("Cron Expression \"" + cronExpression + "\" is invalid.");
+            }
+        }
+        return cronExpression;
+    }
+
+    // Following code is reported as duplicates since all the lines doing same function call.
+    private static String buildCronExpression(BMap<String, BValue> record) {
+        String cronExpression = getStringFieldValue(record, FIELD_SECONDS) + " " +
+                getStringFieldValue(record, FIELD_MINUTES) + " " +
+                getStringFieldValue(record, FIELD_HOURS) + " " +
+                getStringFieldValue(record, FIELD_DAYS_OF_MONTH) + " " +
+                getStringFieldValue(record, FIELD_MONTHS) + " " +
+                getStringFieldValue(record, FIELD_DAYS_OF_WEEK) + " " +
+                getStringFieldValue(record, FIELD_YEAR);
+        return cronExpression.trim();
+    }
+
+    private static String getStringFieldValue(BMap<String, BValue> struct, String fieldName) {
+        if (FIELD_DAYS_OF_MONTH.equals(fieldName) && Objects.isNull(struct.get(FIELD_DAYS_OF_MONTH))) {
+            return "?";
+        } else if (Objects.nonNull(struct.get(fieldName))) {
+            return struct.get(fieldName).stringValue();
+        } else {
+            return "*";
+        }
+    }
+
+    /*
+     * TODO: Runtime validation is done as compiler plugin does not work right now.
+     *       When compiler plugins can be run for the resources without parameters, this will be redundant.
+     */
+    public static void validateService(Service service) throws BLangRuntimeException {
+        Resource[] resources = service.getResources();
+        if (resources.length != 1) {
+            throw new BLangRuntimeException("Invalid number of resources found in service \'" + service.getName()
+                    + "\'. Task service should include only one resource.");
+        }
+        Resource resource = resources[0];
+
+        if (RESOURCE_ON_TRIGGER.equals(resource.getName())) {
+            validateOnTriggerResource(resource.getResourceInfo());
+        } else {
+            throw new BLangRuntimeException("Invalid resource function found: " + resource.getName()
+                    + ". Expected: \'" + RESOURCE_ON_TRIGGER + "\'.");
+        }
+    }
+
+    private static void validateOnTriggerResource(FunctionInfo functionInfo) {
+        if (functionInfo.getRetParamTypes().length != 1 || functionInfo.getRetParamTypes()[0] != BTypes.typeNull) {
+            throw new BLangRuntimeException("Invalid resource function signature: \'"
+                    + RESOURCE_ON_TRIGGER + "\' should not return a value.");
+        }
+        // TODO: Validate input parameters
```

Comment:
Do we need this TODO here? If yes, pls create a github issue to track this and add the details here

Reply:
TODO is not needed.